kotlin-eclipse-ui/src/org/jetbrains/kotlin/ui/editors/codeassist/KotlinCompletionProcessor.java

```diff
@@ -191,19 +200,24 @@ public Boolean invoke(Name name) {
     }
     
     private Collection<ICompletionProposal> generateTemplateProposals(ITextViewer viewer, int offset, String identifierPart) {
-        List<String> contextTypeIds = KotlinApplicableTemplateContext.getApplicableContextTypeIds(viewer, 
-                EditorUtil.getFile(editor), offset - identifierPart.length());
-        
         List<ICompletionProposal> proposals = new ArrayList<ICompletionProposal>();
-        IRegion region = new Region(offset - identifierPart.length(), identifierPart.length());
-        Image templateIcon = JavaPluginImages.get(JavaPluginImages.IMG_OBJS_TEMPLATE);
         
-        List<Template> templates = KotlinApplicableTemplateContext.getTemplatesByContextTypeIds(contextTypeIds);
-        for (Template template : templates) {
-            if (template.getName().startsWith(identifierPart)) {
-                TemplateContext templateContext = createTemplateContext(region, template.getContextTypeId());
-                proposals.add(new TemplateProposal(template, templateContext, region, templateIcon));
+        IFile file = EditorUtil.getFile(editor);
+        if (file != null) {
+            List<String> contextTypeIds = KotlinApplicableTemplateContext.getApplicableContextTypeIds(viewer, file, offset - identifierPart.length());
+
```

Comment:
To imporove code readability you can reduce nested blocks by checking file for null first

```java
IFile file = EditorUtil.getFile(editor);
if (file == null) {
  return Collections.emptyList();
}

List<ICompletionProposal> proposals = new ArrayList<ICompletionProposal>();
...
```